### PR TITLE
Feature/core layer documentation

### PR DESCRIPTION
**Pull request by adamcaak**
Documentation code

### CODE REVIEW

File: src/BookIt.Services.Subscriptions.Core/Domain/Events/SubscriptionCanceled.cs [hunk 1]
```diff
@@ -6,7 +6,10 @@ namespace BookIt.Services.Subscriptions.Core.Domain.Events
     public class SubscriptionCanceled : IDomainEvent
```
Comment (StarostaSebastian) [inline src/BookIt.Services.Subscriptions.Core/Domain/Events/SubscriptionCanceled.cs:6]: Subscription canceled Event (above this class)

**Comment by StarostaSebastian** [comment]
I will make some corrections

**Comment by StarostaSebastian** [comment]
I change SemiAnnualSubscription to FortnightSubscription(2 weeks)

File: src/BookIt.Services.Subscriptions.Api/Startup.cs [hunk 2]
```diff
@@ -15,12 +18,19 @@ public Startup(IConfiguration configuration)
             Configuration = configuration;
         }
 
-
+        /// <summary>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Api/Startup.cs:21]: Wrong. Remove this

File: src/BookIt.Services.Subscriptions.Api/Startup.cs [hunk 3]
```diff
@@ -15,12 +18,19 @@ public Startup(IConfiguration configuration)
...
+        /// Method which a configure the application service
+        /// </summary>
+        /// <param name="services"></param>
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
         }
-
+       /// <summary>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Api/Startup.cs:29]: Wrong. Remove this

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs [hunk 4]
```diff
@@ -7,33 +7,61 @@
...
             SetType();
             Activate();
             SetDates(SystemClock.Instance);
         }
-
+        /// <summary>
+        /// Sets subscribstion expired
+        /// Deactivate subscription and starts subscription expired event
+        /// </summary>
+        /// <param name="clock">Current date time</param>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs:47]: Wrong. Change to: "Instance of a NodaTime clock."

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs [hunk 5]
```diff
@@ -55,17 +87,24 @@ protected virtual void SetId(Guid id)
...
+        /// <summary>
+        /// Sets subscription inactive
+        /// </summary>
         protected void Deactivate()
         {
             IsActive = false;
         }
-
+        /// <summary>
+        /// Sets subscription state
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs:105]: sets subscription **type**. State can be confused with 'isActive'

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs [hunk 6]
```diff
@@ -55,17 +87,24 @@ protected virtual void SetId(Guid id)
...
+        /// </summary>
+        /// <param name="subscriptionType">Subscription type</param>
         protected virtual void ChangeSubscriptionState(SubscriptionTypes subscriptionType)
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs:108]: Can you please rename this method to ChangeSubscriptionType? On second thought it makes more sense.

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs [hunk 7]
```diff
@@ -97,11 +136,19 @@ protected virtual void ChangeSubscriptionState(SubscriptionTypes subscriptionTyp
...
-
+        /// <summary>
+        /// Sets subscription type
+        /// </summary>
         protected abstract void SetType();
-
+        /// <summary>
+        /// Sets subscription dates
+        /// </summary>
+        /// <param name="clock">Current date time</param>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs:146]: Like above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs [hunk 8]
```diff
@@ -97,11 +136,19 @@ protected virtual void ChangeSubscriptionState(SubscriptionTypes subscriptionTyp
...
         protected abstract void SetDates(IClock clock);
-
+        /// <summary>
+        /// Returning type of subscription type
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscription.cs:149]: Can be romoved for now

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/AnnualSubscription.cs [hunk 9]
```diff
@@ -4,25 +4,41 @@
 
 namespace BookIt.Services.Subscriptions.Core.Domain.Models.Subscriptions
 {
+    /// <summary>
+    /// Sealed annual subscription domain model
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/AnnualSubscription.cs:8]: We don't need to explicitly say its sealed. Just Annual subscription domain model

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/AnnualSubscription.cs [hunk 10]
```diff
@@ -4,25 +4,41 @@
...
+        /// </summary>
         protected override void SetType()
         {
             Type = SubscriptionTypes.Annual.ToString();
         }
-
+        /// <summary>
+        /// Sets dates subscription
+        /// </summary>
+        /// <param name="clock">Current date time</param>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/AnnualSubscription.cs:41]: As above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/AnnualSubscription.cs [hunk 11]
```diff
@@ -4,25 +4,41 @@
...
+        /// <summary>
+        /// Sets subscription type
+        /// </summary>
         protected override void SetType()
         {
             Type = SubscriptionTypes.Annual.ToString();
         }
-
+        /// <summary>
+        /// Sets dates subscription
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/AnnualSubscription.cs:39]: Sets subscription start and end dates

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/FortnightSubscription.cs [hunk 12]
```diff
@@ -0,0 +1,48 @@
+﻿using BookIt.Services.Subscriptions.Core.Domain.Events;
+using NodaTime;
+using System;
+
+namespace BookIt.Services.Subscriptions.Core.Domain.Models.Subscriptions
+{
+    /// <summary>
+    /// Sealed fortnight subscription domain model
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/FortnightSubscription.cs:8]: As above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/FortnightSubscription.cs [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+        }
+        /// <summary>
+        /// Sets subscritipon type
+        /// </summary>
+        protected override void SetType()
+        {
+            Type = SubscriptionTypes.Fortnight.ToString();
+        }
+        /// <summary>
+        /// Sets dates subscription
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/FortnightSubscription.cs:39]: As above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/FortnightSubscription.cs [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        /// </summary>
+        /// <param name="clock">Current date time</param>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/FortnightSubscription.cs:41]: As above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/MonthlySubscription.cs [hunk 15]
```diff
@@ -4,25 +4,41 @@
 
 namespace BookIt.Services.Subscriptions.Core.Domain.Models.Subscriptions
 {
+    /// <summary>
+    /// Sealed monthly subscription domain model
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/MonthlySubscription.cs:8]: As above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/MonthlySubscription.cs [hunk 16]
```diff
@@ -4,25 +4,41 @@
...
+        /// <summary>
+        /// Sets subscription type
+        /// </summary>
         protected override void SetType()
         {
             Type = SubscriptionTypes.Monthly.ToString();
         }
-
+        /// <summary>
+        /// Sets subscription dates
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/MonthlySubscription.cs:39]: As above

File: src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/MonthlySubscription.cs [hunk 17]
```diff
@@ -4,25 +4,41 @@
...
+        /// </summary>
+        /// <param name="clock">Current date time</param>
```
Comment (Cichyy7) [inline src/BookIt.Services.Subscriptions.Core/Domain/Models/Subscriptions/MonthlySubscription.cs:41]: As above

File: tests/Unit/BookIt.Services.Subscriptions.Core.Unit.Tests/Domain/Models/SemiAnnualSubscriptionTests.cs [hunk 18]
```diff
@@ -18,12 +18,12 @@ public class SemiAnnualSubscriptionTests
         [Test]
```
Comment (Cichyy7) [inline tests/Unit/BookIt.Services.Subscriptions.Core.Unit.Tests/Domain/Models/SemiAnnualSubscriptionTests.cs:18]: Just a question here. Did you run tests to check if all succeeded? If not, please do that.

**Pull request merged**